### PR TITLE
Add Ctype global state debug printers

### PR DESCRIPTION
**Pull request by ncik-roberts**
This PR adds a way of printing some of `Ctype`'s global state (including current level), and adds the debug printer to the set of printers installed by `tools/debug_printers`.

Output from `ocamldebug`:

```ocaml
(ocd) p Ctype.global_state
Ctype.global_state: Ctype.global_state =
  {current_level = 2; nongen_level = 2; global_level = 2}
```

### CODE REVIEW

**Comment by ncik-roberts** [comment]
I haven't figured out how to request review in the normal way yet (I think I might have to be given write access first?) but I nominate @ccasin to take a look.

File: typing/ctype.ml
Comment (ccasin) [inline typing/ctype.ml:5727]: This makes `Ctype` depend on `Oprint` and `Outcometree`, so the build is broken (at least the makefile one - I didn't try dune).

You can fix this by running `make depend` and committing the updated `.depend` file.  But this feels like a somewhat heavy-weight solution.  Perhaps just print directly instead:
```ocaml
  let record_fields =
     ...
  in
  let print_fields fmt fields =
    List.iter (fun (s,r) -> Format.fprintf fmt "%s = %d;@;" s !r) fields
  in
  Format.fprintf fmt "@[<1>{@;%a}@]" print_fields record_fields
```

**Review by ccasin — CHANGES_REQUESTED** [review]
I've made a couple suggestions, and in particular there's an issue with the build, but the idea seems good.

File: typing/ctype.ml
Comment (ncik-roberts) [reply typing/ctype.ml:5706]: I don’t think this works. When you use the debugger to print a value, it marshals that value from the ocaml process to the debugger process, and then runs the printer code. So I believe taking this suggestion would always print the initial value of levels `(0,0,0)`. (Though I’ll try this shortly to confirm.)

Comment (ccasin) [reply typing/ctype.ml:5706]: You're right!  (I just tried it.)

File: typing/ctype.ml
Comment (ncik-roberts) [reply typing/ctype.ml]: I agree, it seems worse to introduce this dependency than just printing the fields directly (thanks for the suggestion). The output is now:

```
(ocd) p Ctype.global_state
Ctype.global_state: Ctype.global_state =
  { current_level = 2; nongen_level = 2; global_level = 2; }
```

**Review by ccasin — APPROVED** [review]
Looks good now - thanks!  Will merge when the tests pass.

**Pull request merged**
